File: eco-core/core-backend/src/main/java/com/willfp/eco/internal/drops/DropManager.java
package com.willfp.eco.internal.drops;

import com.willfp.eco.core.EcoPlugin;
import com.willfp.eco.core.config.updating.ConfigUpdater;
import lombok.Getter;
import lombok.experimental.UtilityClass;
import org.jetbrains.annotations.NotNull;

@UtilityClass
public final class DropManager {
    @Getter
    private DropQueueType type = DropQueueType.COLLATED;

    @ConfigUpdater
    public static void update(@NotNull final EcoPlugin plugin) {
        type = plugin.getConfigYml().getBool("use-fast-collated-drops") ? DropQueueType.COLLATED : DropQueueType.STANDARD;
    }
}

File: eco-core/core-backend/src/main/kotlin/com/willfp/eco/internal/drops/DropManager.kt
package com.willfp.eco.internal.drops

import com.willfp.eco.core.EcoPlugin
import com.willfp.eco.core.config.updating.ConfigUpdater

object DropManager {
    var type = DropQueueType.COLLATED

    @ConfigUpdater
    @JvmStatic
    fun update(plugin: EcoPlugin) {
        type = if (plugin.configYml.getBool("use-fast-collated-drops")) DropQueueType.COLLATED else DropQueueType.STANDARD
    }
}